### PR TITLE
build(gha): use devservices for kafka

### PR DESCRIPTION
**Pull request by billyvg**
This changes acceptance tests to run kafka and zookeeper as well. The env vars for these hosts were also incorrect (if using devservices).

### CODE REVIEW

**Comment by billyvg** [comment]
Previous test logs: https://github.com/getsentry/sentry/runs/1162842577#step:16:183

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -218,7 +218,7 @@ jobs:
         - name: Start devservices
           run: |
             sentry init
-            sentry devservices up postgres redis clickhouse snuba
+            sentry devservices up
```
Comment (billyvg) [inline .github/workflows/acceptance.yml:221]: Hmm should we be explicit here?

Comment (joshuarli) [reply .github/workflows/acceptance.yml:221]: Yes.

```
Service `asdf` is not known or not enabled.

Services that are available:
postgres
relay
zookeeper
redis
kafka
snuba
clickhouse
```

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -125,9 +125,9 @@ jobs:
         PYTEST_ADDOPTS: "--reruns 5"
 
         # services configuration
-        SENTRY_KAFKA_HOSTS: kafka:9093
```
Comment (joshuarli) [inline .github/workflows/acceptance.yml:128]: Sorry, I didn't catch this originally.

File: .github/workflows/acceptance.yml [hunk 3]
```diff
@@ -276,9 +276,9 @@ jobs:
...
-        SENTRY_ZOOKEEPER_HOSTS: zookeeper:2182
-        SENTRY_REDIS_HOST: redis
+        SENTRY_KAFKA_HOSTS: 127.0.0.1:9092
+        SENTRY_ZOOKEEPER_HOSTS: sentry_zookeeper:2181
+        SENTRY_REDIS_HOST: sentry_redis
```
Comment (joshuarli) [inline .github/workflows/acceptance.yml:281]:
```suggestion
```

This isn't needed, I don't think.

Comment (joshuarli) [reply .github/workflows/acceptance.yml:281]: I think we can remove this across the board.

File: .github/workflows/python.yml [hunk 4]
```diff
@@ -226,10 +226,10 @@ jobs:
       PYTEST_SENTRY_ALWAYS_REPORT: no
 
       # services configuration
-      SENTRY_REDIS_HOST: redis
+      SENTRY_REDIS_HOST: sentry_redis
       DATABASE_URL: postgresql://postgres:postgres@localhost/sentry
-      SENTRY_KAFKA_HOSTS: localhost:9092
-      SENTRY_ZOOKEEPER_HOSTS: localhost:2181
+      SENTRY_KAFKA_HOSTS: 127.0.0.1:9092
+      SENTRY_ZOOKEEPER_HOSTS: sentry_zookeeper:2181
```
Comment (joshuarli) [inline .github/workflows/python.yml:232]:
```suggestion
      SENTRY_ZOOKEEPER_HOSTS: 127.0.0.1:2181
```

And in other areas.

File: src/sentry/conf/server.py [hunk 5]
```diff
@@ -1484,6 +1484,7 @@ def create_partitioned_queues(name):
     },
     "zookeeper": {
         "image": "confluentinc/cp-zookeeper:5.1.2",
+        "ports": {"2181/tcp": 2181},
```
Comment (joshuarli) [inline src/sentry/conf/server.py:1487]: 🤔 This is just purely for annotative purposes which could be improved, but thanks for making it consistent.

File: .github/workflows/acceptance.yml (hunk 3 above)
Comment (billyvg) [reply .github/workflows/acceptance.yml:281]: Yeah I think you're right, since this has been wrong

File: .github/workflows/python.yml (hunk 4 above)
Comment (billyvg) [reply .github/workflows/python.yml]: @joshuarli This doesn't work :< https://github.com/getsentry/sentry/pull/20986/checks?check_run_id=1167768037#step:11:1092

Comment (joshuarli) [reply .github/workflows/python.yml]: Hmmmm, weird, let me try and understand why first.

**Comment by joshuarli** [comment]
`%3|1600984251.330|FAIL|rdkafka#producer-1| [thrd:127.0.0.1:9092/bootstrap]: 127.0.0.1:9092/bootstrap: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT, 30 identical error(s) suppressed)` can be observed running locally w/o kafka and with pytest `-s`. I'm not yet sure as to why this is not fatal.

As to where... this occurs after subbing `"asdf"` for `"default"` in `KAFKA_CLUSTERS` and running `pytest -k test_errors_query tests/acceptance/test_organization_events_v2.py` which store_events:

```
ey:235: in test_errors_query
    assert_no_errors=False,
src/sentry/testutils/cases.py:679: in store_event
    stored_event = Factories.store_event(*args, **kwargs)
src/sentry/testutils/factories.py:532: in store_event
    event = manager.save(project_id)
src/sentry/utils/metrics.py:193: in inner
    return f(*args, **kwargs)
src/sentry/event_manager.py:493: in save
    _track_outcome_accepted_many(jobs)
src/sentry/utils/metrics.py:193: in inner
    return f(*args, **kwargs)
src/sentry/event_manager.py:779: in _track_outcome_accepted_many
    category=job["category"],
src/sentry/utils/outcomes.py:52: in track_outcome
    outcomes_publisher = KafkaPublisher(settings.KAFKA_CLUSTERS[outcomes["cluster"]])
E   KeyError: 'default'
```

But anyway, it's apparent that acceptance tests don't need kafka, so going to remove those env vars. The fixes to them do apply to kafka via devservices though.

**Comment by joshuarli** [comment]
That said it seems devservices kafka is unfit for py3 snuba backend which does need it for a few tests... I think we need something like what I mentioned here: https://github.com/getsentry/sentry/pull/21098#issuecomment-702461434

**Comment by billyvg** [comment]
Thanks for diving into this @joshuarli - I'm gonna close this PR for now, lmk if there are things you want me to cherry pick from this.

**Comment by joshuarli** [comment]
> Thanks for diving into this @joshuarli - I'm gonna close this PR for now, lmk if there are things you want me to cherry pick from this.

@billyvg I threw up https://github.com/getsentry/sentry/pull/21236, but shortly after I remembered you were working on some pretty big changes and that's what you meant by this.

**Pull request closed without merging**